pkg/kube/kubehttp_test.go

```diff
@@ -9,48 +9,26 @@ import (
 	"github.com/weaveworks/weave-gitops/pkg/kube"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	apiruntime "k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
-var testClustername = "test-cluster"
-var testenv *envtest.Environment
-var cfg *rest.Config
-var testclient client.Client
-var err error
-var scheme *apiruntime.Scheme
-var k kube.Kube
-var ns corev1.Namespace
-
-var _ = BeforeSuite(func() {
-	testenv = &envtest.Environment{CRDDirectoryPaths: []string{"../../manifests/crds"}}
-
-	scheme = kube.CreateScheme()
-	cfg, err = testenv.Start()
-
-	Expect(err).NotTo(HaveOccurred())
-
-	ns = corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{Name: kube.WeGONamespace},
-	}
-
-	testclient, err = client.New(cfg, client.Options{Scheme: scheme})
-	Expect(err).NotTo(HaveOccurred())
-
-	err = testclient.Create(context.Background(), &ns)
-	Expect(err).NotTo(HaveOccurred())
-})
-
-var _ = BeforeEach(func() {
-	k = &kube.KubeHTTP{
-		Client:      testclient,
-		ClusterName: testClustername,
-	}
-})
-
 var _ = Describe("KubeHTTP", func() {
+	var (
+		namespace *corev1.Namespace
+		err       error
+	)
+
+	BeforeEach(func() {
+		namespace = &corev1.Namespace{}
+		namespace.Name = "kube-test-" + randStringRunes(5)
```

Comment:
This could be shortened to...

```go
namespace = &corev1.Namespace{Name: "kube-test-" + randStringRunes(5)}
```

Or, if you were getting rid of `randStringrunes`...

```go
import "k8s.io/apimachinery/pkg/util/rand"

namespace = &corev1.Namespace{Name: "kube-test-" + rand.String(5)}
```